classes/class-truelayer-gateway.php

```diff
@@ -180,6 +183,16 @@ public function process_payment( $order_id ) {
 
 		$build_test_url = Truelayer_Helper_Hosted_Payment_Page_URL::build_hosted_payment_page_url( $order_id );
 
+                if( 'EPP' === $epp_enabled ) {
+                        $encoded_url = base64_encode( rawurlencode( home_url( '/wc-api/TrueLayer_Redirect/' ) ) );
+                        $encoded_payment_id = base64_encode( $truelayer_payment_id );
+                        $encoded_payment_token = base64_encode( $truelayer_payment_token );
+                        return array(
+                                'result'   => 'success',
+                                'redirect' => '#truelayer:' . $encoded_url . ':' . $encoded_payment_id . ':' . $encoded_payment_token, // phpcs:ignore
```

Comment:
I'm not sure imploding/exploding with `:` as a delimiter is the best option here. Would using named fragments not suffice?

eg this will generate a url safe set of fragments:

```
'redirect' => '#' . http_build_query([
  'tl_url' => home_url(...),
  'tl_id' => $truelayer_payment_id,
  'tl_token' => $truelayer_payment_token
]);
```

then on the frontend it can easily be read using `URLSearchParams` and the `get` method.

Reply:
Ye thats a good way to solve it, the original implementation followed a similar approach we had in a different case that was slightly different. Made some slight modifications to your suggestion. We rather pass the entire URL for the redirect as the data in the anchor tag, and parse it as a URL in the frontend. Then we extract the search params and get the data we need to pass to the Payment function from your package using the URLSearchParams. You can see the changes in the last commits and if you have any more feedback or suggestions they are welcome.